core/cog-view.c

```diff
@@ -396,3 +437,20 @@ cog_view_get_use_key_bindings(CogView *self)
     CogViewPrivate *priv = cog_view_get_instance_private(COG_VIEW(self));
     return priv->use_key_bindings;
 }
+
+/**
+ * cog_view_get_viewport: (get-property viewport)
+ * @self: A view.
+ *
+ * Gets viewport where the view is attached.
+ *
+ * Returns: (transfer none) (nullable): The viewport what the view belongs.
```

Comment:
This should be `(transfer full)` because the value returned from `g_weak_ref_get()` is also `(transfer full)`. A reference is added to the returned value when the weak reference is fetched to prevent the object to be freed while being used by the caller.

```suggestion
 * Returns: (transfer full) (nullable): The viewport what the view belongs.
```